docs/blog/2021-09-27-crash.md

```diff
@@ -0,0 +1,328 @@
+		  "format": "%Y-%m-%d",
+		  "subtype": "naive_time",
+		  "begin": "1999-01-01",
+		  "end": "2199-31-12"
+		}
+	  }
+       }
+   }
+}
+```
+
+This gets me the following error which is again misplaced at the end of the input, and not exactly understandable. The same happens if I select a date format of `"%H"` and bounds of `0` to `23`.
+
+```console
+Error: Unable to open the namespace
+
+Caused by:
+    0: at file 7_datetime/datetime.json
+    1: Failed to parse collection
+    2: input is not enough for unique date and time at line 16 column 1
+```
+
+I believe since the time is not constrained in any way by the input, we should just issue a warning and generate an unconstrained time instead, so the user will at least get *some* data. Interestingly, seconds seem to be optional, so `%H:%M` works.
+
+\#8 More, if I use `naive_date` instead, but make the minimum `0-0-0`, we get the technically correct but still mis-spanned:
+
+```console
+Error: Unable to open the namespace
+
+Caused by:
+    0: at file 8_endofdays/endofdays.json
+    1: Failed to parse collection
+    2: input is out of range at line 16 column 1s
+```
+
+For the record, the error is on line 11.
+
+\#9 Now we let `date_time` have some rest and go on to [`categorical`](https://www.getsynth.com/docs/content/string#categorical). Having just one variant with a weight of `0` will actually trigger an `unreachable` error:
+
+```json synth
+{
+    "type": "array",
+    "length": 1,
+    "content": {
+        "type": "object",
+        "cat": {
+            "type": "string",
+            "categorical": {
+                "empty": 0
+            }
+        }
+    }
+}
+```
+
+Well, the code thinks we should not be able to reach it. Surprise!
+
+```console
+thread 'main' panicked at 'internal error: entered unreachable code', /home/andre/projects/synth/core/src/schema/content/categorical.rs:82:9
+note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
+```
+
+For the record, this is the first internal error I was able to uncover so far. Given this success with categorical strings, it was natural to look if `one_of` could be similarly broken, but the generator just chose the one variant despite its `0.0` weight.
+
+\#10 Unsupported types on import
+
+Databases can sometimes contain strange things, and so far the support is in beta, so it was expected that I would find types for which we currently don't implement import. This includes JSON for mysql and postgres, the mysql spatial datatypes as well as postgres' geometric types, user-defined enumerations, postgres' network address types, postgres arrays (soon only nested ones), etc.
+
+The way to reproduce that is to create a table with a field of the type, e.g. here with mysql
+
+```sql
+CREATE TABLE IF NOT EXISTS json (
+    data JSON
+);
+
+DELETE FROM json;
+
+INSERT INTO json (data) VALUES ('{ "a": ["b", 42] }');
+```
+
+Now call `synth import fail --from mysql://<user>:<password>@<host>:<port>/<database>` to get
```

Comment:
The `fail` namespace name may be misleading here?

Reply:
Ok, I'll use `jsonnamespace` instead.